### PR TITLE
[Search] Add 2023-11-01 support

### PR DESCRIPTION
**Pull request by mattmsft**
**Related command**

az search

**Description**

* Add 2023-11-01 support
* Add semantic search support

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Search

File: src/azure-cli/azure/cli/command_modules/search/commands.py [hunk 1]
```diff
@@ -46,10 +46,14 @@ def load_command_table(self, _):
         g.command('list', 'list_by_resource_group')
         g.show_command('show', 'get')
         g.command('delete', 'delete', confirmation=True)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/search/commands.py:48]: You should remove these two lines if you want to use the `delete` and `show` commands from aaz directly

File: src/azure-cli/azure/cli/command_modules/search/commands.py [hunk 2]
```diff
@@ -46,10 +46,14 @@ def load_command_table(self, _):
...
-        g.generic_update_command('update', supports_no_wait=True, custom_func_name='update_search_service', setter_name='begin_create_or_update', setter_arg_name='service')
-        g.custom_command('create', 'create_search_service', supports_no_wait=True)
+        # g.generic_update_command('update', supports_no_wait=True, custom_func_name='update_search_service', setter_name='begin_create_or_update', setter_arg_name='service')
+        # g.custom_command('create', 'create_search_service', supports_no_wait=True)
         g.wait_command('wait')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/search/commands.py:51]: Also need to remove the `wait` command

File: src/azure-cli/azure/cli/command_modules/search/aaz/latest/search/service/__cmd_group.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command_group(
+    "search service",
+)
+class __CMDGroup(AAZCommandGroup):
+    """Demo
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/search/aaz/latest/search/service/__cmd_group.py:18]: Please update the help message of this command group.

File: src/azure-cli/azure/cli/command_modules/search/aaz/latest/search/__cmd_group.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command_group(
+    "search",
+)
+class __CMDGroup(AAZCommandGroup):
+    """Demo
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/search/aaz/latest/search/__cmd_group.py:18]: Also this one

File: src/azure-cli/azure/cli/command_modules/search/custom.py [hunk 5]
```diff
@@ -17,6 +19,121 @@ def _get_resource_group_location(cli_ctx, resource_group_name):
...
+    def pre_instance_update(self, instance):
+        from azure.cli.core.aaz import has_value
+        args = self.ctx.args
+        if has_value(args.auth_options):
+            # clean up current auth_options
+            instance.properties.auth_options = {}
+
+
 # pylint: disable=too-many-locals
 def create_search_service(cmd, resource_group_name, search_service_name, sku, location=None, partition_count=0,
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/search/custom.py:138]: You can clean up the unused custom function as wee

**Comment by kairu-ms** [comment]
Please fix the CI [issues](https://dev.azure.com/azclitools/public/_build/results?buildId=97893&view=logs&j=9ccb9437-6abb-5c1a-b909-dd7aa8d5d916&t=da342424-93bf-5216-b53c-16ba29f2062e)

File: src/azure-cli/azure/cli/command_modules/search/commands.py (hunk 1 above)
Comment (mattmsft) [reply src/azure-cli/azure/cli/command_modules/search/commands.py:48]: done

File: src/azure-cli/azure/cli/command_modules/search/commands.py (hunk 2 above)
Comment (mattmsft) [reply src/azure-cli/azure/cli/command_modules/search/commands.py:51]: done

File: src/azure-cli/azure/cli/command_modules/search/aaz/latest/search/service/__cmd_group.py (hunk 3 above)
Comment (mattmsft) [reply src/azure-cli/azure/cli/command_modules/search/aaz/latest/search/service/__cmd_group.py:18]: done

File: src/azure-cli/azure/cli/command_modules/search/aaz/latest/search/__cmd_group.py (hunk 4 above)
Comment (mattmsft) [reply src/azure-cli/azure/cli/command_modules/search/aaz/latest/search/__cmd_group.py:18]: done

File: src/azure-cli/azure/cli/command_modules/search/custom.py (hunk 5 above)
Comment (mattmsft) [reply src/azure-cli/azure/cli/command_modules/search/custom.py:138]: done

**Comment by mattmsft** [comment]
@kairu-ms any idea how to change your active subscription for azdev test? az account set --subscription xxx doesn't seem to work

**Comment by kairu-ms** [comment]
> @kairu-ms any idea how to change your active subscription for azdev test? az account set --subscription xxx doesn't seem to work

You can modify subscription id in this file directly 
image

**Comment by mattmsft** [comment]
> > @kairu-ms any idea how to change your active subscription for azdev test? az account set --subscription xxx doesn't seem to work
> 
> You can modify subscription id in this file directly image

worked, thanks

**Review by kairu-ms — APPROVED** [review]
Please format your PR title. You can follow this [guideline](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#format-pr-title)

**Pull request merged**